pkg/datagatherer/k8s/generic.go

```diff
@@ -110,9 +112,41 @@ func (g *DataGatherer) Fetch() (interface{}, error) {
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+	// Redact Secret data
+	err = redactList(list)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
 	return list, nil
 }
 
+func redactList(list *unstructured.UnstructuredList) error {
+	// In principal we could only redact the list if it's kind is SecretList or
+	// a generic mixed List, however the test suite does not set the list kind
+	// and it is safer to always check for Secrets.
+	// Iterate over the items in the list.
+	for i := range list.Items {
+		// Determine the kind of items in case this is a generic 'mixed' list.
+		gvks, _, err := scheme.Scheme.ObjectKinds(list.Items[i].DeepCopyObject())
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		redactItem := false
+		for _, gvk := range gvks {
+			// If this item is a Secret then we need to redact it.
+			if gvk.Kind == "Secret" {
+				redactItem = true
+				break
+			}
+		}
+		// Redcat the Secret by overwriting its data.
+		if redactItem {
```

Comment:
Why do this outside of the loop? Can this be done with `redactItem` gets set to true above instead?

Reply:
Ah yeah, not sure why I did it that way...